src/encoder.rs

```diff
@@ -127,14 +127,40 @@ fn encode_arg(arg: &OscType) -> Result<(Option<Vec<u8>>, char)> {
             for (i, v) in x.iter().enumerate() {
                 bytes[i + 4] = *v;
             }
-            Ok((Some(bytes), 'b'))
+            Ok((Some(bytes), 'b'.to_string()))
         }
-        OscType::Time(ref x, ref y) => Ok((Some(encode_time_tag(*x, *y)), 't')),
-        OscType::Midi(ref x) => Ok((Some(vec![x.port, x.status, x.data1, x.data2]), 'm')),
-        OscType::Color(ref x) => Ok((Some(vec![x.red, x.green, x.blue, x.alpha]), 'r')),
-        OscType::Bool(ref x) => if *x { Ok((None, 'T')) } else { Ok((None, 'F')) },
-        OscType::Nil => Ok((None, 'N')),
-        OscType::Inf => Ok((None, 'I')),
+        OscType::Array(ref x) => {
+            let mut bytes = vec![0u8; x.len()];
+            let mut type_tags = String::from("[");
+            for v in x.iter() {
+                match encode_arg(v).unwrap() {
+                    (Some(other_bytes), other_type_tags) => {
+                        bytes.extend(other_bytes);
+                        type_tags.extend(other_type_tags.chars());
```

Comment:
Not sure if this more idiomatic but `String`s provide `push`: `type_tags.push_str(&other_type_tags);`.